proofs/pfedit.ml

```diff
@@ -177,7 +179,10 @@ let refine_by_tactic env sigma ty tac =
       iraise (e, info)
   in
   (** Plug back the retrieved sigma *)
-  let sigma = Proof.in_proof prf (fun sigma -> sigma) in
+  let (goals,stack,shelf,given_up,sigma) = Proof.proof prf in
+  assert (stack = []);
+(*  if shelf <> [] then CErrors.user_err (str "Tactic not supposed to shelve goals.");*)
+  if given_up <> [] then CErrors.user_err (str "Tactic not supposed to given up goals.");
```

Comment:
Grammar nit: I'd instead say "Giving up goals in tactics in terms is unsupported".  (Or, at the very least, "given" should be "give" in your message.)

Reply:
The problem is that this function is not used only by tactic-in-term. It is also used e.g. in `setoid_rewrite`.

Anyway, I'm more in the mood of supporting `ltac:(shelve)` in `refine`, as you also asked, than to forbid it. So this error message is more probably going to disappear.